conf/sage.config

```diff
@@ -0,0 +1,58 @@
+params {
+  config_profile_description = 'The Sage Bionetworks profile'
+  config_profile_contact = 'Bruno Grande (@BrunoGrandePhD)'
+  config_profile_url = 'https://github.com/Sage-Bionetworks-Workflows'
+}
+
+process {
+
+  cpus   = { check_max( 1    * slow(task.attempt), 'cpus'   ) }
+  memory = { check_max( 6.GB * task.attempt,       'memory' ) }
+  time   = { check_max( 24.h * task.attempt,       'time'   ) }
+
+  errorStrategy = { task.exitStatus in [143,137,104,134,139,247] ? 'retry' : 'finish' }
+  maxRetries    = 5
+  maxErrors     = '-1'
+
+  // Process-specific resource requirements
+  withLabel:process_low {
+    cpus   = { check_max( 4     * slow(task.attempt),  'cpus'   ) }
+    memory = { check_max( 12.GB * task.attempt,        'memory' ) }
+    time   = { check_max( 24.h  * task.attempt,        'time'   ) }
+  }
+  withLabel:process_medium {
+    cpus   = { check_max( 12    * slow(task.attempt), 'cpus'   ) }
+    memory = { check_max( 36.GB * task.attempt,       'memory' ) }
+    time   = { check_max( 48.h  * task.attempt,       'time'   ) }
+  }
+  withLabel:process_high {
+    cpus   = { check_max( 24    * slow(task.attempt), 'cpus'   ) }
+    memory = { check_max( 72.GB * task.attempt,       'memory' ) }
+    time   = { check_max( 96.h  * task.attempt,       'time'   ) }
+  }
+  withLabel:process_long {
+    time   = { check_max( 192.h  * task.attempt,   'time'   ) }
+  }
+  withLabel:process_high_memory {
+    memory = { check_max( 200.GB * task.attempt,   'memory' ) }
+  }
+
+}
+
+aws {
+  region = "us-east-1"
+}
+
+params {
+  igenomes_base = 's3://sage-igenomes/igenomes'
+  max_memory    = 512.GB
+  max_cpus      = 64
+  max_time      = 168.h  // One week
+}
+
+// Function to slow the increase of the resource multipler
+// as attempts are made. The rationale is that some CPUs
+// don't need to be increased as fast as memory.
+def slow(attempt, factor = 2) {
+  return Math.ceil( attempt / factor) as int
+}
```

Comment:
I like this level of customization, but why not directly changing
```
cpus   = { check_max( 4     * slow(task.attempt),  'cpus'   ) }
```
to
```
cpus   = { check_max( 2     * task.attempt,  'cpus'   ) }
```

Reply:
In this particular case, I don't want to decrease the initial number of allocated CPU cores. For example:

```
cpus   = { check_max( 4     * slow(task.attempt),  'cpus'   ) }
```

would result in these CPU requests over five attempts: 4, 4, 8, 8, and 12. 

Whereas:

```
cpus   = { check_max( 2     * task.attempt,  'cpus'   ) }
```

would result in these CPU requests over five attempts: 2, 2, 4, 4, and 6. 

The former "progression" matches what I would like to see. In other words, try increasing the memory first without changing the number of CPU cores (since memory is more commonly the limiting factor), and if you need to re-run a third time, then try increasing the number of CPU cores.